### PR TITLE
fix(block): add accessible label for slotted controls

### PR DESCRIPTION
**Pull request by driskull**
**Related Issue:** #8037

## Summary

- Add aria-labelledby to internal control container for enhanced accessibility.

### CODE REVIEW

**Comment by driskull** [comment]
@geospatialem could you help test this out?

File: packages/calcite-components/src/components/block/block.tsx [hunk 1]
```diff
@@ -370,7 +370,7 @@ export class Block
           headerContent
         )}
         {hasControl ? (
-          <div class={CSS.controlContainer}>
+          <div aria-describedby={IDS.header} class={CSS.controlContainer}>
```
Comment (geospatialem) [inline packages/calcite-components/src/components/block/block.tsx:373]:
```suggestion
          <div aria-labelledby={IDS.header} class={CSS.controlContainer}>
```

**Comment by driskull** [comment]
> I'm not sure the best path forward, as it seems browsers and AT is still catching up to support with shadow DOM. Maybe we just account for JAWS here, and mention NVDA is out of scope? Otherwise there are some hacky loop holes, maybe using a live region? WDYT?

Yeah we might want to mention NVDA is out of scope until it catches up with shadow dom. The next best option would be aria-live but I don't think thats a great solution.

File: packages/calcite-components/src/components/block/block.tsx [hunk 2]
```diff
@@ -370,7 +370,7 @@ export class Block
           headerContent
         )}
         {hasControl ? (
-          <div class={CSS.controlContainer}>
+          <div aria-labelledby={IDS.header} class={CSS.controlContainer}>
```
[the remark this answers is not in the document]
Comment (driskull) [reply packages/calcite-components/src/components/block/block.tsx:373]: I think its just the title and icon.

https://github.com/Esri/calcite-design-system/blob/ac33969cdc860ab3b76d2a137685ee6d9d7202f9/packages/calcite-components/src/components/block/block.tsx#L342-L347

**Review by geospatialem — APPROVED** [review]
LGTM! 🚀 

Before merging, does the PR summary scope need to be updated to fit the label over description?

**Pull request merged**